[9.0] [Files Service] Additional validation for mime-type allowances (#234828)

tsullivan opened the pull request:
# Backport

This will backport the following commits from `main` to `9.0`:
 - [[Files Service] Additional validation for mime-type allowances (#234828)](https://github.com/elastic/kibana/pull/234828)

<!--- Backport version: 10.0.2 -->

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--BACKPORT [{"author":{"name":"Tim Sullivan","email":"tsullivan@users.noreply.github.com"},"sourceCommit":{"committedDate":"2025-09-23T19:27:06Z","message":"[Files Service] Additional validation for mime-type allowances (#234828)\n\n## Summary\n\nSummarize your PR. If it involves visual changes include a screenshot or\ngif.\n\n\n### Checklist\n\nCheck the PR satisfies following conditions. \n\nReviewers should verify this PR satisfies this list as well.\n\n- ~~[ ] Any text added follows [EUI's writing\nguidelines](https://elastic.github.io/eui/#/guidelines/writing), uses\nsentence case text and includes [i18n\nsupport](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)~~\n- ~~[ ]\n[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)\nwas added for features that require explanation or tutorials~~\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios\n- ~~[ ] If a plugin configuration key changed, check if it needs to be\nallowlisted in the cloud and added to the [docker\nlist](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)~~\n- ~~[ ] This was checked for breaking HTTP API changes, and any breaking\nchanges have been approved by the breaking-change committee. The\n`release_note:breaking` label should be applied in these situations.~~\n- ~~[ ] [Flaky Test\nRunner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was\nused on any tests changed~~~\n- ~~[ ] The PR description includes the appropriate Release Notes\nsection, and the correct `release_note:*` label is applied per the\n[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)~~\n- [x] Review the [backport\nguidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)\nand apply applicable `backport:*` labels.\n\n### Identify risks\n\nDoes this PR introduce any risks? For example, consider risks like hard\nto test bugs, performance regression, potential of data loss.\n\nDescribe the risk, its severity, and mitigation for each identified\nrisk. Invite stakeholders and evaluate how to proceed before merging.\n\n- [ ] [See some risk\nexamples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)\n- [ ] ...\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"cf5e9704688c732fb404da7c4b884591482e0b00","branchLabelMapping":{"^v9.2.0$":"main","^v(\\d+).(\\d+).\\d+$":"$1.$2"}},"sourcePullRequest":{"labels":["release_note:skip","backport:all-open","v9.2.0"],"title":"[Files Service] Additional validation for mime-type allowances","number":234828,"url":"https://github.com/elastic/kibana/pull/234828","mergeCommit":{"message":"[Files Service] Additional validation for mime-type allowances (#234828)\n\n## Summary\n\nSummarize your PR. If it involves visual changes include a screenshot or\ngif.\n\n\n### Checklist\n\nCheck the PR satisfies following conditions. \n\nReviewers should verify this PR satisfies this list as well.\n\n- ~~[ ] Any text added follows [EUI's writing\nguidelines](https://elastic.github.io/eui/#/guidelines/writing), uses\nsentence case text and includes [i18n\nsupport](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)~~\n- ~~[ ]\n[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)\nwas added for features that require explanation or tutorials~~\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios\n- ~~[ ] If a plugin configuration key changed, check if it needs to be\nallowlisted in the cloud and added to the [docker\nlist](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)~~\n- ~~[ ] This was checked for breaking HTTP API changes, and any breaking\nchanges have been approved by the breaking-change committee. The\n`release_note:breaking` label should be applied in these situations.~~\n- ~~[ ] [Flaky Test\nRunner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was\nused on any tests changed~~~\n- ~~[ ] The PR description includes the appropriate Release Notes\nsection, and the correct `release_note:*` label is applied per the\n[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)~~\n- [x] Review the [backport\nguidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)\nand apply applicable `backport:*` labels.\n\n### Identify risks\n\nDoes this PR introduce any risks? For example, consider risks like hard\nto test bugs, performance regression, potential of data loss.\n\nDescribe the risk, its severity, and mitigation for each identified\nrisk. Invite stakeholders and evaluate how to proceed before merging.\n\n- [ ] [See some risk\nexamples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)\n- [ ] ...\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"cf5e9704688c732fb404da7c4b884591482e0b00"}},"sourceBranch":"main","suggestedTargetBranches":[],"targetPullRequestStates":[{"branch":"main","label":"v9.2.0","branchLabelMappingKey":"^v9.2.0$","isSourceBranch":true,"state":"MERGED","url":"https://github.com/elastic/kibana/pull/234828","number":234828,"mergeCommit":{"message":"[Files Service] Additional validation for mime-type allowances (#234828)\n\n## Summary\n\nSummarize your PR. If it involves visual changes include a screenshot or\ngif.\n\n\n### Checklist\n\nCheck the PR satisfies following conditions. \n\nReviewers should verify this PR satisfies this list as well.\n\n- ~~[ ] Any text added follows [EUI's writing\nguidelines](https://elastic.github.io/eui/#/guidelines/writing), uses\nsentence case text and includes [i18n\nsupport](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)~~\n- ~~[ ]\n[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)\nwas added for features that require explanation or tutorials~~\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios\n- ~~[ ] If a plugin configuration key changed, check if it needs to be\nallowlisted in the cloud and added to the [docker\nlist](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)~~\n- ~~[ ] This was checked for breaking HTTP API changes, and any breaking\nchanges have been approved by the breaking-change committee. The\n`release_note:breaking` label should be applied in these situations.~~\n- ~~[ ] [Flaky Test\nRunner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was\nused on any tests changed~~~\n- ~~[ ] The PR description includes the appropriate Release Notes\nsection, and the correct `release_note:*` label is applied per the\n[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)~~\n- [x] Review the [backport\nguidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)\nand apply applicable `backport:*` labels.\n\n### Identify risks\n\nDoes this PR introduce any risks? For example, consider risks like hard\nto test bugs, performance regression, potential of data loss.\n\nDescribe the risk, its severity, and mitigation for each identified\nrisk. Invite stakeholders and evaluate how to proceed before merging.\n\n- [ ] [See some risk\nexamples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)\n- [ ] ...\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"cf5e9704688c732fb404da7c4b884591482e0b00"}}]}] BACKPORT-->

tsullivan committed:
[Files Service] Additional validation for mime-type allowances (#234828)

## Summary

Summarize your PR. If it involves visual changes include a screenshot or
gif.

### Checklist

Check the PR satisfies following conditions.

Reviewers should verify this PR satisfies this list as well.

- ~~[ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)~~
- ~~[ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials~~
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- ~~[ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)~~
- ~~[ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.~~
- ~~[ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed~~~
- ~~[ ] The PR description includes the appropriate Release Notes
section, and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)~~
- [x] Review the [backport
guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing)
and apply applicable `backport:*` labels.

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
(cherry picked from commit cf5e9704688c732fb404da7c4b884591482e0b00)

# Conflicts:
#	src/platform/plugins/shared/files/server/routes/file_kind/create.ts
#	src/platform/plugins/shared/files/server/routes/file_kind/helpers.ts
#	src/platform/plugins/shared/files/server/routes/integration_tests/routes.test.ts
#	src/platform/plugins/shared/files/server/routes/public_facing/download.ts
#	src/platform/plugins/shared/files/tsconfig.json

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/342894)
* Commit: 6614a64178c9aa71fe757e0ccbc9f05822be5286

### Failed CI Steps
* [Jest Integration Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f)
* [Jest Integration Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a08-4c75-8bc0-15b536f7e141) Jest Integration Tests #<span></span>3 / File HTTP API public download allows public download with matching file extension
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf8-4fdb-b80c-33ae08d934b7) Jest Integration Tests #<span></span>3 / File HTTP API public download allows public download with matching file extension
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a05-4de0-8803-da13bd5a68e3) Jest Integration Tests #<span></span>3 / File HTTP API public download does not leak information in public download error messages
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf5-40d6-9a8d-afd14080683d) Jest Integration Tests #<span></span>3 / File HTTP API public download does not leak information in public download error messages
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a56-4775-b8fb-d0c05e0dc1e2) Jest Integration Tests #<span></span>3 / File HTTP API public download handles case insensitive extensions in public download
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf7-4280-9a36-dc07a2060c6e) Jest Integration Tests #<span></span>3 / File HTTP API public download handles case insensitive extensions in public download
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a04-406f-ab58-863df367a1fd) Jest Integration Tests #<span></span>3 / File HTTP API public download handles Unicode filenames in public download validation
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf8-42cb-b818-575e8024034d) Jest Integration Tests #<span></span>3 / File HTTP API public download handles Unicode filenames in public download validation
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a07-4c86-af19-b80e3970eb1d) Jest Integration Tests #<span></span>3 / File HTTP API public download it downloads a publicly shared file
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf3-4256-98de-b1b4c53b2991) Jest Integration Tests #<span></span>3 / File HTTP API public download it downloads a publicly shared file
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a04-4daf-a823-cddfaffb0419) Jest Integration Tests #<span></span>3 / File HTTP API public download prevents MIME type manipulation through URL filename
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf7-4f90-bb94-20660c38e216) Jest Integration Tests #<span></span>3 / File HTTP API public download prevents MIME type manipulation through URL filename
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a07-4660-b766-18841289160d) Jest Integration Tests #<span></span>3 / File HTTP API public download validates file extension in public download
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5e84-4cb9-88ab-d323b268b817) Jest Integration Tests #<span></span>3 / File HTTP API public download validates file extension in public download
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a05-4ff0-b299-dc8a8280684b) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should allow download with matching file extension
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5e85-4ee1-9be0-9f580c7e3e27) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should allow download with matching file extension
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a04-4eb8-81c3-68731b7a74bb) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should be case insensitive for file extensions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf8-4e91-b391-9bbecf438cd3) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should be case insensitive for file extensions
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a57-423d-8883-a29b9de950b1) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should handle multiple extensions correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf6-41ac-a732-9ad5d91b2255) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should handle multiple extensions correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a04-43c3-a48c-743e0612d37d) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should reject download with mismatched file extension
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf7-4879-b36a-e7ce4b16cb3f) Jest Integration Tests #<span></span>3 / File kind HTTP API file extension validation on download should reject download with mismatched file extension
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a07-4e0e-abe7-1ec601e069ef) Jest Integration Tests #<span></span>3 / File kind HTTP API MIME type validation should allow files with permitted MIME types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5cf8-41fd-a0d1-09c8b37f5018) Jest Integration Tests #<span></span>3 / File kind HTTP API MIME type validation should allow files with permitted MIME types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#019978fb-7b01-45b2-8006-ee1002a5b21f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/019978fb-7b01-45b2-8006-ee1002a5b21f/artifacts/0199791e-2a56-407d-b08a-917fe9b11dfb) Jest Integration Tests #<span></span>3 / File kind HTTP API MIME type validation should reject files with forbidden MIME types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/342894#0199791e-4015-4857-a906-12069c2635c3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/342894/jobs/0199791e-4015-4857-a906-12069c2635c3/artifacts/01997930-5e84-4d07-8d51-619acf620730) Jest Integration Tests #<span></span>3 / File kind HTTP API MIME type validation should reject files with forbidden MIME types

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/342894","number":342894,"commit":"6614a64178c9aa71fe757e0ccbc9f05822be5286"}],"number":342894}
buildkite-pr-comment-->

tsullivan committed:
Fix integration tests

These tests are not applicable because they would only pass if the branch includes changes from https://github.com/elastic/kibana/pull/217361

tsullivan committed:
Merge branch '9.0' into backport/9.0/pr-234828

tsullivan committed:
restore more integration tests

kc13greiner reviewed (approved):
Backport LGTM!
